presto-main/src/main/java/com/facebook/presto/server/SerializedPageWriteListener.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.execution.buffer.SerializedPage;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import javax.servlet.AsyncContext;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+
+import java.io.IOException;
+import java.util.Queue;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+import static java.util.Objects.requireNonNull;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+
+public class SerializedPageWriteListener
+        implements WriteListener
+{
+    private final Logger log = Logger.get(SerializedPageWriteListener.class);
+
+    private static final int BUFFER_SIZE = 4 * 1024;
```

Comment:
Why such a large buffer needed? You are writing only 13 bytes in it.

Reply:
Why do you need 4kb buffer? Why 13 bytes is not enough?

Reply:
> Why do you need 4kb buffer? Why 13 bytes is not enough?

I see,  this is not needed after the change, I forget to decrease the size.